packages/common/gotrue.ts

```diff
@@ -33,10 +33,29 @@ if (debug) {
   }
 }
 
+let previousOperation: Promise<any> = Promise.resolve()
```

Comment:
Consider adding type to the 'previousOperation' variable. This will improve the type safety of the code. [medium]